### PR TITLE
Upgrade the WordPress coding standards.

### LINKED ISSUE #7293: Upgrade the WordPress coding standards to 2.3.0
## Feature Description

Upgrade the WordPress coding standards to 2.3.0

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

While working #6609, I noticed the WordPress coding standards were out of date. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7293

### CODE REVIEW

File: includes/Discovery.php [hunk 1]
```diff
@@ -221,7 +221,7 @@ public function print_open_graph_metadata() {
 		$metadata = $this->get_open_graph_metadata();
 
 		foreach ( $metadata as $name => $value ) {
-			printf( '<meta property="%s" content="%s" />', esc_attr( $name ), esc_attr( $value ) );
+			printf( '<meta property="%s" content="%s" />', esc_attr( $name ), esc_attr( $value ) ); // phpcs:ignore WordPressVIPMinimum.Security.ProperEscapingFunction.notAttrEscAttr
```
Comment (miina) [inline includes/Discovery.php]: Hm, why didn't it pass here? How would it need to be then for this rule to pass?

Comment (spacedmonkey) [reply includes/Discovery.php]: It is a new rule added in 2.3.0 coding standards. It expects `esc_attr` to be used as part of a html tag and not part of `printf`. Basically this is a false positive.

Comment (miina) [reply includes/Discovery.php]: Could the rules be adjusted to not give this false positive?

Comment (spacedmonkey) [reply includes/Discovery.php]: Seems to be a known issue - https://github.com/Automattic/VIP-Coding-Standards/pull/668

Comment (miina) [reply includes/Discovery.php]: Ah, thanks for flagging, so it should get updated again once the fix gets released. Could we add this as a comment to the code (E.g. something like `/* Disable reason: known false positive bug, see https://github.com/Automattic/VIP-Coding-Standards/pull/668 */`)?

Comment (spacedmonkey) [reply includes/Discovery.php]: It was merged in 2.3.1. I have updated now remove this. However, there are other new rules.

File: tests/phpunit/tests/AMP/Optimization.php [hunk 2]
```diff
@@ -99,6 +101,8 @@ public function test_get_optimizer_configuration_no_ssr() {
 
 		$transformers = [
 			AmpBoilerplate::class,
+			AmpBoilerplateErrorHandler::class,
```
Comment (spacedmonkey) [inline tests/phpunit/tests/AMP/Optimization.php:104]: This is a somewhat unreleated changed, as the amp plugin was update on composer update.

File: includes/Widgets/Stories.php [hunk 3]
```diff
@@ -115,7 +115,7 @@ public function widget( $args, $instance ) {
 		];
 
 		$story_args = [
-			'posts_per_page' => min( absint( $instance['number_of_stories'] ), 20 ),
+			'posts_per_page' => min( absint( $instance['number_of_stories'] ), 20 ), // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
```
Comment (spacedmonkey) [inline includes/Widgets/Stories.php:118]: This is a new false positive. WordPress.WP.PostsPerPage.posts_per_page_posts_per_page

Comment (miina) [reply includes/Widgets/Stories.php:118]: Hm, so it's falsely warning about the `posts_per_page` being too high? Is there an issue open for this, too?

Comment (spacedmonkey) [reply includes/Widgets/Stories.php:118]: No, there is no issue, which is why I ignored. It is not a hard coded value, so the linter can't check it.

Comment (miina) [reply includes/Widgets/Stories.php:118]: Hm, should this be reported then? Since previously it didn't trigger false positive for it.

**Review by miina — APPROVED** [review]

**Pull request merged**